editor/src/main/java/org/pentaho/pms/ui/util/Splash.java

```diff
@@ -91,12 +91,30 @@ protected Splash( Display display, Shell splashShell, boolean isStartup ) {
     shell.open();
 
     if ( isStartup ) { //show splash for 5s on startup
-      long endTime = System.currentTimeMillis() + 5000;
-      while ( !shell.isDisposed() && endTime > System.currentTimeMillis() ) {
-        if ( isMacOS() && !display.readAndDispatch() ) {
+
+      // Start timer in a separate thread so that it isn't blocked and that it doesn't block clicks to close the
+      // splash early
+      new Thread( () -> {
+        try {
+          Thread.sleep( 5000 ); // Sleep for 5 seconds
+        } catch ( InterruptedException e ) {
+          e.printStackTrace();
```

Comment:
Please address this code smell that sonar caught

```
 } catch ( InterruptedException e ) {
 e.printStackTrace();
 ....
```

You prob just use the logger for this class, I'm not sure if we want to

- ignore the exception and log it, since it's just a splash screen
- or since `Runnable` can't throw exceptions, you can use [Callable](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Callable.html) which can be wrapped inside a `Thread` , some examples https://blog.knoldus.com/threading-and-futures-in-java/

Reply:
yeah, I am thinking I can just log a warning that the timer thread's sleep was interrupted, and then it should proceed to close